src/demo-app/datepicker/datepicker-demo.ts

```diff
@@ -40,4 +41,49 @@ export class DatepickerDemo {
 
   onDateInput = (e: MatDatepickerInputEvent<Date>) => this.lastDateInput = e.value;
   onDateChange = (e: MatDatepickerInputEvent<Date>) => this.lastDateChange = e.value;
+
+  // pass custom header component type as input
+  customHeader = CustomHeader;
+}
+
+// Custom header component for datepicker
+@Component({
+  selector: 'custom-header',
+  template: `
+      <div>
+        <button (click)="previousClicked('year')">&lt;&lt;</button>
+          <button (click)="previousClicked('month')">&lt;</button>
+        {{periodLabel}}
+        <button (click)="nextClicked('month')">&gt;</button>
+          <button (click)="nextClicked('year')">&gt;&gt;</button>
+      </div>
+  `
+})
+export class CustomHeader {
```

Comment:
Instead of using <any> below, can you add a type parameter to this class `CustomHeader<T>`

Reply:
done in 715e413cd27043b92fb8ba96f741dce1f6e96afd